src/proxy/cache/itemStore.js

```diff
@@ -28,32 +28,43 @@ import itemPreloaderFactory from 'taoQtiTest/runner/proxy/cache/itemPreloader';
 /**
  * The default number of items to store
  */
-var defaultConfig = {
+const defaultConfig = {
+    itemTTL: 0,
     maxSize: 10,
     preload: false
 };
 
 /**
  * Create an item store
- * @param {Object} [options]
- * @param {Number} [options.maxSize = 10] - the store limit
- * @param {Boolean} [options.preload] - do we preload items when storing them
- * @param {String} [options.testId] - the unique identifier of the test instance, required if preload is true
+ * @param {object} [options]
+ * @param {number} [options.maxSize = 10] - the store limit
+ * @param {boolean} [options.preload] - do we preload items when storing them
+ * @param {string} [options.testId] - the unique identifier of the test instance, required if preload is true
  *
  * @returns {itemStore}
  */
 export default function itemStoreFactory(options) {
-    var config = _.defaults(options || {}, defaultConfig);
+    const config = _.defaults(options || {}, defaultConfig);
 
     //in memory storage
-    var getStore = function getStore() {
-        return store('item-cache', store.backends.memory);
+    const getStore = () => store('item-cache', store.backends.memory);
+
+    //maintain an index to resolve existence synchronously:
+    // - the array allows to manage the cache in order, it is useful to remove the first entries
+    // - the map allows to quickly check each cache entry, and store some internal metadata like the timestamp
+    let list = [];
+    const index = new Map();
```

Comment:
I don't love the idea of managing the Array and the Map in parallel. The scope is luckily quite limited, but it's still a potential trap for someone refactoring in future. Do you think it can be done using only the Map and a comparison function at the moment of removing the earliest item? Or will that be worse for readability and performance?

Reply:
Map objects can keep the insertion order. However, there is no way to access them by position. Eventually, we could rely on the iteration protocol to access the first position, but it may have unexpected behaviour since we may have a race condition that will disturb the insertion order (items are added in burst without waiting for the previous to be set).

We could indeed maintain a position value for each item in the map, and then search for the lower one when trimming. It will have a low impact on performance considering the small amount of entries. But it will also have some complexity.

I was also considering implementing an ordered map in a separate module, offering an API to manage it. But I lacked time and I made the choice to go straight to the essential.